moped-database/migrations/1669080983051_unique_subcomponent_and_component/up.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE moped_proj_components_subcomponents ADD CONSTRAINT unique_component_and_subcomponent UNIQUE ("project_component_id", "subcomponent_id");
```

Comment:
This constraint prevents us from adding already existing subcomponents associated with a component again. It enables us to update `is_deleted` on conflict so we can update a previously soft deleted subcomponent back in by flipping `is_deleted` from `true` to `false` again.